tox.ini

```diff
@@ -54,11 +54,42 @@ commands =
     pip-audit {env:EXCLUDES:}
 
 [testenv:docs,docs-py{39,310,311,312}]
-description = Build documentation
+description = Build documentation by building the examples, checking the links and
```

Comment:
Since @AdrianSosic  wished for it, here are some details on the workflows that can be used now - I think having a thread to discuss this makes sense.
1. There is no difference when you call the `tox` command for doc building as before, including the options for `force` and `ignore_examples`.
2. The ideal workflow would be to have a locally built documentation that was once created via the full comment and then just run the `quickbuild` command as required. This should be the fastest way to quickly re-generate the documentation.
3. Only running `quickbuild` without creating the examples first requires the `force` flag, since we have an explicit link to the examples folder which is only created when calling the script for creating the examples (also, I'm getting the weird `is_mc` error again, but this only happens if I do a quickbuild with no previously built documentation). This might however still be sufficient to just check some visuals (although I highly recommend option 2 and see no reason why this should not be the default option).

Reply:
Hi @AVHopp, thanks for the overview. I think splitting the monolithic script into more atomic parts is a good first step toward improving our doc-building pipeline 👍🏼 

However, I'm coming more from a workflow-oriented perspective here, and I can't really say at this point if the new tox commands will ever be used. Instead, I rather have a list of workflows that I'd ideally like to have covered, which I think is not yet the case? In essence, I see the following three realistic workflows only (let me know if you have others):
1. Building the documentation completely, with all necessary checks involved. Either for local inspection or finally in the CI. This is possible simply via `tox -e docs` ✅ 
2. To me, most important workflow when working on the docs: When writing/updating a non-example section (e.g. user guide), I just want to get this part rendered asap, typically including a link check. For this, I don't need to see any examples, nor do I even want to process them with the "-e" flag, which still takes very long initially. My current "manual" workflow for this (which is still way faster than everything we currently provide via tox) is to manually delete the content of the example folder first, pretending there are no examples yet. Currently not possible with the given setup ❌. But couldn't we get this easily running in a similar same way?
3. Sometimes, I run 2. but don't want a linkcheck. I guess this is currently possible via `tox -e docs-quickbuild -- --force`? ❓ But I still wonder why it takes comparably long, i.e. ˜50 seconds on my machine. Is this related to the `autosummary` caching problem you showed me?

Reply:
But why don't just keep an already built documentation and update it via `quickbuild` whenever you make a change? I still do not see any argument against this, since this can easily be integrated into your workflow since this just requires a **single** full doc building.
Regardless of that, here are some comments on your other points:
2. This should be possible by providing another command that just runs `linkcheck` and `quickbuild`, either with the `force` option or via an additional command that just creates the `examples/examples` folder via e.g. `mkdir`. 
3. If you do not have a pre-built documentation, it needs to re-create everything, so this just takes some time. If you already have a built documentation, then this is exactly the workflow that I described under "my" 2. which is significantly faster (although I also observed that it sometimes seemed to take ~50 seconds instead of ~20).

Reply:
The answer why I don't pre-build the examples / keep an already built doc is easy: convenience. Sometimes, I have to delete build folders, or I run stuff in a different worktree, or similar, so the doc is not guaranteed to exist and even if it dit, it would likely be outdated. Pre-building before doing quickbuild is simply inconvenient. It takes me a second to delete the content of the example folder versus several minutes to run the build. I think our pipeline should provide a more elegant solution than my quick hack of deleting stuff.

So rethinking what I think is actually needed, I think would come up with the following setup (comments welcome):
* A single `docs` command (I don't really get the purpose of `quickbuild` if I can't use it without running `docs` upfront) that by default takes the "fastest" route. That is, if necessary, it builds stuff from scratch, otherwise uses caching as much as it can.
* A flag than bypasses caching and makes sure stuff is fully rebuild
* A "force" option to bypass linkchecks
* An "ignore examples" option that makes the call behave as if there were no example in place. That is, is doesn't touch the examples at all, not even pre-processing them. I think linkcheck can be easily fixed for this case by explicitly adding the example folder to a whitelist

That would pretty much cover it all! Or is there anything missing? No need for several tox environments, and no need to switch commands / have commands that can fail if executed in the wrong order.

Reply:
Working on making stuff more streamlined. Will use this thread to describe what I implement and how to streamline everything.

Reply:
This was adjusted by now - see the details that I shared in our chat as well as the description of the docstring. Is there anything missing/not working as expected?

Reply:
moving on with the PR, lets do outstanding things at a next iteration
`ze way is ze target`